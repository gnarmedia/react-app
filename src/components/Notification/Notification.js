/**
 * @file Notification component
 */

import React from 'react'
import PropTypes from 'prop-types'

/**
 * Renders Notification component.
 *
 * @example
 * <Notification type="success" />
 * <Notification type="error" />
 *
 * @param {*} props - Props.
 *
 * @returns {Function} Notification component.
 */
const Notification = ({ type, isClosable, children }) => {
  let className = 'notification'

  if (type === 'info') {
    className += ' is-info'
  } else if (type === 'success') {
    className += ' is-success'
  } else if (type === 'warning') {
    className += ' is-warning'
  } else if (type === 'error') {
    className += ' is-danger'
  }

  return (
    <div className={className}>
      {isClosable && <button type="button" className="delete" />}
      {children}
    </div>
  )
}

Notification.propTypes = {
  type: PropTypes.string,
  isClosable: PropTypes.bool,
  children: PropTypes.node.isRequired,
}

Notification.defaultProps = {
  type: '',
  isClosable: false,
}

export default Notification
