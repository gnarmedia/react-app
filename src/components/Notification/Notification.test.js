/**
 * @file Input component tests
 */

import React from 'react'
import { render } from 'enzyme'

import Notification from './Notification'

describe('index', () => {
  it('should render the Notification component', () => {
    const wrapper = render(<Notification>notification</Notification>)

    expect(wrapper).toMatchSnapshot()
  })

  it('should render the Notification component for info', () => {
    const wrapper = render(<Notification type="info">info</Notification>)

    expect(wrapper).toMatchSnapshot()
  })

  it('should render the Notification component for success', () => {
    const wrapper = render(<Notification type="success">success</Notification>)

    expect(wrapper).toMatchSnapshot()
  })

  it('should render the Notification component for warning', () => {
    const wrapper = render(<Notification type="warning">warning</Notification>)

    expect(wrapper).toMatchSnapshot()
  })

  it('should render the Notification component for error', () => {
    const wrapper = render(<Notification type="error">error</Notification>)

    expect(wrapper).toMatchSnapshot()
  })
})
