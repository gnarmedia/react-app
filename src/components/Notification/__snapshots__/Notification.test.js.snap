// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index should render the Notification component 1`] = `
<div
  class="notification"
>
  notification
</div>
`;

exports[`index should render the Notification component as closable 1`] = `
<div
  class="notification"
>
  <button
    class="delete"
    type="button"
  />
  closable
</div>
`;

exports[`index should render the Notification component for error 1`] = `
<div
  class="notification is-danger"
>
  error
</div>
`;

exports[`index should render the Notification component for info 1`] = `
<div
  class="notification is-info"
>
  info
</div>
`;

exports[`index should render the Notification component for success 1`] = `
<div
  class="notification is-success"
>
  success
</div>
`;

exports[`index should render the Notification component for warning 1`] = `
<div
  class="notification is-warning"
>
  warning
</div>
`;
